src/transaction/log_page_buffer.c

```diff
@@ -1123,123 +773,70 @@ logpb_fix_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, PAGE_FETCH_MODE fetc
   assert ((fetch_mode == NEW_PAGE) || (fetch_mode == OLD_PAGE));
   assert (LOG_CS_OWN_WRITE_MODE (thread_p));
 
-  START_EXCLUSIVE_ACCESS_LOG_PB (rv, thread_p);
-
-  log_bufptr = (LOG_BUFFER *) mht_get (log_Pb.ht, &pageid);
-  if (log_bufptr == NULL)
+  if (pageid == LOGPB_HEADER_PAGE_ID)
     {
-      /* 
-       * Page is not resident. Find a replacement buffer for it
-       */
-
-      stat_page_found = PERF_PAGE_MODE_OLD_LOCK_WAIT;
+      log_bufptr = &log_Pb.header_buffer;
+    }
+  else
+    {
+      log_bufptr = &log_Pb.buffers[logpb_get_log_buffer_index ((int) pageid)];
```

Comment:
@funkydvd 

I'd like to propose to change the line as safe as possible. We may have a bad log pageid for some cases, for instance, log page corruption, bad log record and so on. 

```C
  index = logpb_get_log_buffer_index ((int) pageid);
  if (index is in boundary)
    log_bufptr = ...;
  else
    er_set (...);
    return NULL;
```